packages/@ember/-internals/glimmer/lib/helpers/helper.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ @module ember
+ */
+
+/**
+ Use the `{{helper}}` helper to create contextual helper so
+ that it can be passed around as first-class values in templates.
+
+ ```handlebars
+ {{#let (helper "join-words" "foo" "bar" separator=" ") as |foo-bar|}}
+
+   {{foo-bar}}
+
+   {{!-- these will pass the helper itself into the component, instead of invoking it now --}}
+
+   <MyComponent @helper={{helper foo-bar "baz"}} />
+
+   {{!-- these will yield the helper itself ("contextual helper"), instead of invoking it now --}}
+
+   {{yield foo-bar}}
```

Comment:
```suggestion
   {{!-- this will pass the helper itself into the component, instead of invoking it now --}}
   <MyComponent @helper={{helper foo-bar "baz"}} />

   {{!-- this will yield the helper itself ("contextual helper"), instead of invoking it now --}}
   {{yield foo-bar}}
```